naga/tests/example_wgsl.rs

```diff
@@ -1,55 +1,51 @@
 #![cfg(feature = "wgsl-in")]
 
 use naga::{front::wgsl, valid::Validator};
-use std::{fs, path::PathBuf};
+use std::{ffi::OsStr, fs, path::Path};
 
 /// Runs through all example shaders and ensures they are valid wgsl.
 #[test]
 pub fn parse_example_wgsl() {
-    let read_dir = match PathBuf::from(env!("CARGO_MANIFEST_DIR"))
-        .join("examples")
-        .read_dir()
-    {
-        Ok(iter) => iter,
-        Err(e) => {
-            log::error!("Unable to open the examples folder: {:?}", e);
-            return;
-        }
-    };
-    for example_entry in read_dir {
-        let read_files = match example_entry {
-            Ok(dir_entry) => match dir_entry.path().read_dir() {
-                Ok(iter) => iter,
-                Err(_) => continue,
-            },
-            Err(e) => {
-                log::warn!("Skipping example: {:?}", e);
-                continue;
-            }
+    let example_path = Path::new(env!("CARGO_MANIFEST_DIR"))
+        .parent()
+        .unwrap()
+        .join("examples");
+
+    println!("Looking for examples in {}", example_path.display());
+
+    let mut example_paths = Vec::new();
+    for example_entry in walkdir::WalkDir::new(example_path) {
+        let Ok(dir_entry) = example_entry else {
+            continue;
         };
-        for file_entry in read_files {
-            let shader = match file_entry {
-                Ok(entry) => match entry.path().extension() {
-                    Some(ostr) if ostr == "wgsl" => {
-                        println!("Validating {:?}", entry.path());
-                        fs::read_to_string(entry.path()).unwrap_or_default()
-                    }
-                    _ => continue,
-                },
-                Err(e) => {
-                    log::warn!("Skipping file: {:?}", e);
-                    continue;
-                }
-            };
-
-            let module = wgsl::parse_str(&shader).unwrap();
-            //TODO: re-use the validator
-            Validator::new(
-                naga::valid::ValidationFlags::all(),
-                naga::valid::Capabilities::all(),
-            )
-            .validate(&module)
-            .unwrap();
+
+        if !dir_entry.file_type().is_file() {
+            continue;
+        }
+
+        let path = dir_entry.path();
+
+        if path.extension().map(OsStr::to_string_lossy).as_deref() == Some("wgsl") {
+            example_paths.push(path.to_path_buf());
         }
     }
+
+    assert_ne!(example_paths.len(), 0, "No examples found!");
+
+    println!("Found {} examples", example_paths.len());
+
+    for example_path in example_paths {
+        println!("\tParsing {}", example_path.display());
+
+        let shader = fs::read_to_string(&example_path).unwrap();
+
+        let module = wgsl::parse_str(&shader).unwrap();
+        //TODO: re-use the validator
```

Comment:
thought: It'd be nice to have a separate message for parsing vs. validation. Doesn't have to happen now, tho.

Reply:
Maybe, I didn't really think about it beyond fixing enumeration :)